java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -1753,6 +1755,48 @@ public Table explode(int index) {
     return new Table(explode(nativeHandle, index));
   }
 
+  /**
+   * Explodes a list column's elements and includes a position column.
+   *
+   * Any list is exploded, which means the elements of the list in each row are expanded into new rows
+   * in the output. The corresponding rows for other columns in the input are duplicated. A position
+   * column is added that has the index inside the original list for each row. Example:
+   * <code>
+   * [[5,10,15], 100],
+   * [[20,25],   200],
+   * [[30],      300],
+   * returns
+   * [0,   5,    100],
+   * [1,   10,   100],
+   * [2,   15,    100],
+   * [0,   20,    200],
+   * [1,   25,    200],
+   * [0,   30,    300],
+   * </code>
+   *
+   * Nulls and empty lists propagate in different ways depending on what is null or empty.
+   * <code>
+   * [[5,null,15], 100],
+   * [null,        200],
+   * [[],          300],
+   * returns
+   * [0,    5,     100],
+   * [1,    null,  100],
+   * [2,    15,    100],
+   * </code>
+   *
+   * Note that null lists are not included in the resulting table, but nulls inside
+   * lists and empty lists will be represented with a null entry for that column in that row.
```

Comment:
I know this is a copy of the comment from cudf, but is this actually true? I don't think that empty lists will be exposed as a null entry for the column and the block above shows the empty list disappearing as well. I believe empty lists not being represented in the output is the desired behavior, but I'd like to hear from @revans2 or @jlowe to be certain.

Reply:
Spark does not discard null entries nor empty lists, e.g.:
```
scala> sql("select posexplode(array(array(5,null,15), null, array()))").show
+---+--------+
|pos|     col|
+---+--------+
|  0|[5,, 15]|
|  1|    null|
|  2|      []|
+---+--------+


scala> sql("select posexplode(array(array(5,null,15), null, array()))").collect
res20: Array[org.apache.spark.sql.Row] = Array([0,WrappedArray(5, null, 15)], [1,null], [2,WrappedArray()])
```

Reply:
Sorry, that was posexplode in isolation on a column.  When applied to a table with other columns the semantics indeed match Spark:
```
>>> df.show()
+--------+---+
|     arr|val|
+--------+---+
|[5,, 15]|100|
|    null|200|
|      []|300|
+--------+---+
>>> df.select(posexplode(df.arr),df.val).show()
+---+----+---+
|pos| col|val|
+---+----+---+
|  0|   5|100|
|  1|null|100|
|  2|  15|100|
+---+----+---+
```

Reply:
I agree with Jason that this comment is correct. And I add some null/empty entities into test to ensure cuDF working as the comment describing.

Reply:
I may have been unclear. The issue I see is the wording `empty lists will be represented with a null entry for that column in that row` as an empty list is not represented in the output as there is no 300 in this example. My suggestion would be to change the comment to  say that:
```
* Note that null and empty lists are not included in the resulting table, but nulls inside
* lists will be represented with a null entry for that column in that row.
```